### PR TITLE
Add test case for c5

### PR DESCRIPTION
**Pull request by maximilianpilz**
Add test case for c5.
This includes a switch to MdibVersion based sorting instead of timestamp based sorting and a change of the database layout, as those changes were necessary for the c5 test case.

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Adherence to coding conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Adherence to javadoc conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* config update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* SDCcc executable ran against a test device (if necessary)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

File: sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java [hunk 1]
```diff
@@ -113,12 +134,163 @@ void setUp() throws CertificateException, IOException {
...
+                    messageStorage)) {
+                message.write(String.format(
+                                BASE_MESSAGE_STRING,
+                                "action",
+                                String.format(SEQUENCE_ID_METRIC_BODY_STRING, "9223372036854775807", "1"))
+                        .getBytes(StandardCharsets.UTF_8));
+            }
+            messageStorage.flush();
+
+            final MessageContent messageContent =
```
Comment (belagertem) [inline sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java:212]: This throws an IOException for me. If I use try-with-resources, the exception no longer occurs.

**Review by belagertem — CHANGES_REQUESTED** [review]

File: configuration/config.toml [hunk 2]
```diff
@@ -1,20 +1,20 @@
 [SDCcc]
-CIMode=false
+CIMode=true
 GraphicalPopups=true
 
 [SDCcc.TLS]
-FileDirectory="./configuration"
+FileDirectory="C:/Users/pilzmaxi/Documents/dev/certs"
```
[the remark this answers is not in the document]
Comment (maximilianpilz) [reply configuration/config.toml]: reverted changes in config.toml

File: sdccc/pom.xml [hunk 3]
```diff
@@ -179,7 +179,6 @@
             <groupId>com.draeger.medical</groupId>
             <artifactId>biceps-model</artifactId>
             <version>${revision}${changelist}</version>
-            <scope>test</scope>
```
[the remark this answers is not in the document]
Comment (maximilianpilz) [reply sdccc/pom.xml:182]: This change is no longer needed, I will revert it. It was needed for a previous solution that  didn't push.

File: sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java (hunk 1 above)
Comment (maximilianpilz) [reply sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java]: It is true that there was a connection leak and I fixed it, but I didn't get an IOException here.

File: sdccc/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (maximilianpilz) [reply sdccc/pom.xml]: Reverted

File: sdccc/src/main/java/com/draeger/medical/sdccc/messages/mapping/MdibVersionGroupEntity.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import org.hibernate.annotations.GenericGenerator;
+
+/**
+ * POJO for persisting headers.
```
[the remark this answers is not in the document]
Comment (maximilianpilz) [reply sdccc/src/main/java/com/draeger/medical/sdccc/messages/mapping/MdibVersionGroupEntity.java]: Fixed

**Comment by maximilianpilz** [comment]
@belagertem I deleted one of your comments, because it did no longer apply to any commit in this PR after the git rebase

File: sdccc/src/test/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMultiStateTestTest.java [hunk 5]
```diff
@@ -275,7 +275,18 @@ public void testRequirementR0097GoodOneAcceptableSequenceSeen() throws Exception
                 null,
                 null,
                 buildOperatorContext(
-                        OPERATOR_CONTEXT_DESCRIPTOR_HANDLE2, OPERATOR_CONTEXT_STATE_HANDLE2, BigInteger.TWO));
+                        OPERATOR_CONTEXT_DESCRIPTOR_HANDLE, OPERATOR_CONTEXT_STATE_HANDLE2, BigInteger.TWO));
```
Comment (belagertem) [inline sdccc/src/test/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMultiStateTestTest.java:278]: Why the change?

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java [hunk 6]
```diff
@@ -87,6 +90,98 @@ void setup() {
         this.soapUtil = riInjector.getInstance(SoapUtil.class);
     }
 
+    @Test
+    @TestIdentifier(EnabledTestConfig.BICEPS_C5)
+    @TestDescription("Starting from the initially retrieved mdib, applies each episodic report to the mdib and checks"
+            + " for each AbstractDescriptor contained in a DescriptionModificationReport "
+            + "that it was inserted or deleted or udpdated by changing"
```
Comment (belagertem) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java:97]: typo in _updated_

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java [hunk 7]
```diff
@@ -87,6 +90,98 @@ void setup() {
...
+            + "at least one child or attribute.")
```
Comment (belagertem) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java:98]: preceding whitespace missing

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java [hunk 8]
```diff
@@ -87,6 +90,98 @@ void setup() {
...
+                    var second = mdibHistorian.createNewStorage(sequenceId);
+
+                    for (final Iterator<AbstractReport> iterator = reports.iterator(); iterator.hasNext(); ) {
+                        final AbstractReport report = iterator.next();
+
+                        if (report instanceof DescriptionModificationReport) {
+                            acceptableSequenceSeen.incrementAndGet();
+                            second = mdibHistorian.applyReportOnStorage(second, report);
+
+                            for (var reportPart : ((DescriptionModificationReport) report).getReportPart()) {
```
Comment (belagertem) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java:121]: You could remove the cast and create a variable in the instanceof check

**Comment by belagertem** [comment]
Please update the README section _Which Manipulation is required for which test_.

File: sdccc/src/test/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMultiStateTestTest.java (hunk 5 above)
Comment (maximilianpilz) [reply sdccc/src/test/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMultiStateTestTest.java:278]: this is to make all sequence ids relevant for the test, it was useful when debugging the R0097 test case

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java (hunk 6 above)
Comment (maximilianpilz) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java]: fixed

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java (hunk 7 above)
Comment (maximilianpilz) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java]: fixed

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java (hunk 8 above)
Comment (maximilianpilz) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java]: done

**Comment by maximilianpilz** [comment]
@belagertem README has been adapted regarding the used manipulations

**Review by belagertem — APPROVED** [review]
